src/branchmanager.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+class BranchManager {
```

Comment:
Aquí falta añadir los includes de las clases que usas.

```suggestion
#pragma once

#include "hashtable.hpp"
#include "branch.hpp"

class BranchManager {
```